client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -301,12 +295,27 @@ def _try_fill_write(self) -> None:
         result = self._mtp.finalize()
         if result:
             container_msg_id, mtp_buffer = result
+
+            self._transport.pack(mtp_buffer, self._writer.write)
             for request in self._requests:
                 if isinstance(request.state, Serialized):
-                    request.state.container_msg_id = container_msg_id
+                    request.state = Sent(request.state.msg_id, container_msg_id)
```

Comment:
Any reason to not do this after `drain`? In theory, the data isn't sent until `drain` finishes. Is Telegram responding before this happens?